packages/block-library/src/gallery/edit.js

```diff
@@ -340,6 +344,11 @@ class GalleryEdit extends Component {
 		}
 
 		const imageSizeOptions = this.getImagesSizeOptions();
+		const shouldShowSizeOptions = hasImages && ! isEmpty( imageSizeOptions );
+		// This is needed to fix a separator fence-post issue on mobile.
+		const mobileLinkToProps = shouldShowSizeOptions ?
+			MOBILE_CONTROL_PROPS :
+			MOBILE_CONTROL_PROPS_SEPARATOR_NONE;
```

Comment:
This isn't very nice, but I agree that we need a better general solution for the separators. For this one it's ok